pkg/kube/client.go

```diff
@@ -1274,13 +1274,18 @@ func setServerInfoWithIstiodVersionInfo(serverInfo *version.BuildInfo, istioInfo
 	nParts := len(versionParts)
 	if nParts >= 3 {
 		// The format will be like 1.12.0-016bc46f4a5e0ef3fa135b3c5380ab7765467c1a-dirty-Modified
-		// version is '1.12.0'
-		// revision is '016bc46f4a5e0ef3fa135b3c5380ab7765467c1a-dirty'
-		// status is 'Modified'
-		serverInfo.Version = versionParts[0]
-		serverInfo.GitTag = serverInfo.Version
-		serverInfo.GitRevision = strings.Join(versionParts[1:nParts-1], "-")
+		// version is '1.12.0' || '1.12.0-custom-build'
```

Comment:
is this right?  @howardjohn ptal

Reply:
imo this can enhanced compatibility when a self-compiled version exists. 
@howardjohn  ptal